### PR TITLE
Add go mod vendor check to integration test

### PR DESCRIPTION
**Pull request by ChenYi015**
## Purpose of this PR

**Proposed changes:**

- Add go mod vendor check to integration test

## Change Category

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that could affect existing functionality)
- [ ] Documentation update

### CODE REVIEW

File: .github/workflows/integration.yaml [hunk 1]
```diff
@@ -35,6 +35,14 @@ jobs:
             exit 1
           fi
 
+      - name: Run go mod vendor
+        run: |
+          go mod vendor
+          if ! git diff --quiet; then
+            echo "Please run 'go mod vendor' to make vendored copy of dependencies"
```
Comment (cheyang) [inline .github/workflows/integration.yaml:42]: go mod tidy && go mod vendor

Comment (ChenYi015) [reply .github/workflows/integration.yaml:42]: @cheyang There already is a step for checking `go mod tidy`, see: https://github.com/kubeflow/arena/blob/29b2d6d2c5df6959a05c9de8a81c7436a7820d98/.github/workflows/integration.yaml#L30-L36

Comment (cheyang) [reply .github/workflows/integration.yaml:42]: Ok

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**